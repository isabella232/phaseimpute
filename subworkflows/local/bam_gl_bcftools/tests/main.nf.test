nextflow_workflow {

    name "Test Subworkflow BAM_GL_BCFTOOLS"
    script "../main.nf"
    config "./nextflow.config"

    workflow "BAM_GL_BCFTOOLS"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "subworkflows/bam_gl_bcftools"
    tag "bam_gl_bcftools"

    tag "gawk"
    tag "tabix"
    tag "tabix/bgzip"
    tag "bcftools"
    tag "bcftools/mpileup"
    tag "bcftools/annotate"

    test("Compute genotype likelihood with merging") {
        when {
            workflow {
                """
                input[0] = Channel.fromList([
                    [
                        [id: "NA12878"],
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA12878/NA12878.s.bam", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA12878/NA12878.s.bam.bai", checkIfExist:true),
                    ],
                    [
                        [id: "NA19401"],
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA19401/NA19401.s.bam", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA19401/NA19401.s.bam.bai", checkIfExist:true),
                    ],
                ])
                input[1] = Channel.of([
                        [id: "1000GP", chr: "22"],
                        file(params.pipelines_testdata_base_path + "hum_data/panel/chr22/1000GP.chr22.legend.gz", checkIfExist:true)
                    ]).collect()
                input[2] = Channel.of([
                    [id: "GRCh38"],
                    file(params.pipelines_testdata_base_path + "hum_data/reference_genome/GRCh38.s.fa.gz", checkIfExist:true),
                    file(params.pipelines_testdata_base_path + "hum_data/reference_genome/GRCh38.s.fa.gz.fai", checkIfExist:true),
                ]).collect()
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.versions,
                    workflow.out.vcf_tbi.collect{[
                        it[0],
                        path(it[1]).getFileName().toString(),
                        path(it[2]).getFileName().toString()
                    ] },
                    workflow.out.vcf_tbi.collect{
                        path(it[1]).vcf.summary
                    }
                ).match() }
            )
        }
    }

    test("Compute genotype likelihood with only one input") {
        when {
            workflow {
                """
                input[0] = Channel.fromList([
                    [
                        [id: "NA12878"],
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA12878/NA12878.s.bam", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA12878/NA12878.s.bam.bai", checkIfExist:true),
                    ],
                ])
                input[1] = Channel.of([
                        [id: "1000GP", chr: "22"],
                        file(params.pipelines_testdata_base_path + "hum_data/panel/chr22/1000GP.chr22.legend.gz", checkIfExist:true)
                    ]).collect()
                input[2] = Channel.of([
                    [id: "GRCh38"],
                    file(params.pipelines_testdata_base_path + "hum_data/reference_genome/GRCh38.s.fa.gz", checkIfExist:true),
                    file(params.pipelines_testdata_base_path + "hum_data/reference_genome/GRCh38.s.fa.gz.fai", checkIfExist:true),
                ]).collect()
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.versions,
                    workflow.out.vcf_tbi.collect{[
                        it[0],
                        path(it[1]).getFileName().toString(),
                        path(it[2]).getFileName().toString()
                    ] },
                    workflow.out.vcf_tbi.collect{
                        path(it[1]).vcf.summary
                    }
                ).match() }
            )
        }
    }
}
