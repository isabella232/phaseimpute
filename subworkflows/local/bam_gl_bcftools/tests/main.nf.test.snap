{
    "Compute genotype likelihood with merging": {
        "content": [
            {
                "0": [
                    [
                        {
                            "panel": "1000GP",
                            "chr": "22",
                            "id": "all",
                            "metas": [
                                {
                                    "id": "NA12878",
                                    "panel": "1000GP",
                                    "chr": "22"
                                },
                                {
                                    "id": "NA19401",
                                    "panel": "1000GP",
                                    "chr": "22"
                                }
                            ],
                            "variantcaller": "bcftools"
                        },
                        "all.annotate.vcf.gz:md5,95cf1e61c8d2776a02588952fb67c5d6",
                        "all.annotate.vcf.gz.tbi:md5,e22a83a66d796dca37eab6d75e76f9c3"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0980a298fde671494f5b4e9cb3a6ebc2",
                    "versions.yml:md5,4739962c934731254db8b31083aa0df5",
                    "versions.yml:md5,544ff5e39aa23f18c41587a739beaf34",
                    "versions.yml:md5,74118beefc19065bed2de5d60277be55",
                    "versions.yml:md5,c9d32dd20061f19422ff4d4b6f3fb24d",
                    "versions.yml:md5,c9d32dd20061f19422ff4d4b6f3fb24d"
                ],
                "2": [
                    "NA12878.bcftools_stats.txt:md5,4cacbe4d6dfba4a8035e37611c91923c",
                    "NA19401.bcftools_stats.txt:md5,f91fd586acfdf3845d111a062fbb263f"
                ],
                "multiqc_files": [
                    "NA12878.bcftools_stats.txt:md5,4cacbe4d6dfba4a8035e37611c91923c",
                    "NA19401.bcftools_stats.txt:md5,f91fd586acfdf3845d111a062fbb263f"
                ],
                "vcf_tbi": [
                    [
                        {
                            "panel": "1000GP",
                            "chr": "22",
                            "id": "all",
                            "metas": [
                                {
                                    "id": "NA12878",
                                    "panel": "1000GP",
                                    "chr": "22"
                                },
                                {
                                    "id": "NA19401",
                                    "panel": "1000GP",
                                    "chr": "22"
                                }
                            ],
                            "variantcaller": "bcftools"
                        },
                        "all.annotate.vcf.gz:md5,95cf1e61c8d2776a02588952fb67c5d6",
                        "all.annotate.vcf.gz.tbi:md5,e22a83a66d796dca37eab6d75e76f9c3"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0980a298fde671494f5b4e9cb3a6ebc2",
                    "versions.yml:md5,4739962c934731254db8b31083aa0df5",
                    "versions.yml:md5,544ff5e39aa23f18c41587a739beaf34",
                    "versions.yml:md5,74118beefc19065bed2de5d60277be55",
                    "versions.yml:md5,c9d32dd20061f19422ff4d4b6f3fb24d",
                    "versions.yml:md5,c9d32dd20061f19422ff4d4b6f3fb24d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-06T15:29:03.964528903"
    },
    "Compute genotype likelihood with only one input": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "NA12878",
                            "panel": "1000GP",
                            "chr": "22",
                            "variantcaller": "bcftools"
                        },
                        "NA12878.annotate.vcf.gz:md5,15fdaa69229a6502d2638c467f4341e3",
                        "NA12878.annotate.vcf.gz.tbi:md5,21b8ed54102dc1e3a3596cc080039693"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0980a298fde671494f5b4e9cb3a6ebc2",
                    "versions.yml:md5,4739962c934731254db8b31083aa0df5",
                    "versions.yml:md5,544ff5e39aa23f18c41587a739beaf34",
                    "versions.yml:md5,c9d32dd20061f19422ff4d4b6f3fb24d"
                ],
                "2": [
                    "NA12878.bcftools_stats.txt:md5,4cacbe4d6dfba4a8035e37611c91923c"
                ],
                "multiqc_files": [
                    "NA12878.bcftools_stats.txt:md5,4cacbe4d6dfba4a8035e37611c91923c"
                ],
                "vcf_tbi": [
                    [
                        {
                            "id": "NA12878",
                            "panel": "1000GP",
                            "chr": "22",
                            "variantcaller": "bcftools"
                        },
                        "NA12878.annotate.vcf.gz:md5,15fdaa69229a6502d2638c467f4341e3",
                        "NA12878.annotate.vcf.gz.tbi:md5,21b8ed54102dc1e3a3596cc080039693"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0980a298fde671494f5b4e9cb3a6ebc2",
                    "versions.yml:md5,4739962c934731254db8b31083aa0df5",
                    "versions.yml:md5,544ff5e39aa23f18c41587a739beaf34",
                    "versions.yml:md5,c9d32dd20061f19422ff4d4b6f3fb24d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-06T15:29:22.939785381"
    },
    "Summary_one": {
        "content": [
            [
                "VcfFile [chromosomes=[chr22], sampleCount=1, variantCount=903, phased=false, phasedAutodetect=false]"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T18:40:28.756168568"
    },
    "Summary_merge": {
        "content": [
            [
                "VcfFile [chromosomes=[chr22], sampleCount=2, variantCount=903, phased=false, phasedAutodetect=false]"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T18:35:39.946019792"
    }
}